twoliter/embedded/rpm2img

```diff
@@ -240,6 +240,37 @@ INVENTORY_DATA="$(jq --arg PKG_PREFIX "bottlerocket-" \
 INVENTORY_DATA="$(jq --slurp 'sort_by(.Name)' <<< "${INVENTORY_DATA}" | jq '{"Content": .}')"
 printf "%s\n" "${INVENTORY_DATA}" > "${ROOT_MOUNT}/usr/share/bottlerocket/application-inventory.json"
 
+elf_note() {
+  local input output size offset uint32_t
+  input="${1:?}"
+  output="${2:?}"
+  name="Bottlerocket"
+  name_len="$(( ${#name} + 1 ))"
+  input_len="$(( ${#input} + 1 ))"
+  rm -f "${output}"
+  for n in "${name_len}" "${input_len}" 0 ; do
+    printf -v uint32_t '\\x%02x\\x%02x\\x%02x\\x%02x' \
+      $((n & 255)) $((n >> 8 & 255)) $((n >> 16 & 255)) $((n >> 24 & 255))
+    printf "${uint32_t}" >> "${output}"
+  done
+  echo -en "${name}\x00" >> "${output}"
+  echo -en "${input}\x00" >> "${output}"
+}
+
+# add elf note to relevant binaries
+elf_note "${VARIANT}" "note.elf"
+
+ELF_MAGIC="\x7fELF"
+for some_file in "${ROOT_MOUNT}"/usr/bin/*; do
+  # ensure that the file is not a link to avoid adding duplicate notes
+  if ! readlink "${some_file}" >/dev/null; then
+    # check that the file is an ELF binary
+    if [[ "$(head -c 4 ${some_file})" == "$(echo -e ${ELF_MAGIC})" ]]; then
+      objcopy --add-section .note.bottlerocket.variant="note.elf" "${some_file}"
+    fi
+  fi
+done
```

Comment:
nit: consider factoring the ELF test into an `is_elf()` helper function.

How long does it take to add the note to all binaries? We might need to be smarter about which binaries we touch if it's longer than a couple seconds.